docs/src/spec/impls/iceberg.md

```diff
@@ -0,0 +1,47 @@
+# Lance Iceberg Namespace
+
+**Lance Iceberg Namespace** is an implementation using Apache Iceberg REST Catalog (IRC).
+For more details about IRC, please read the [IRC Specification](https://github.com/apache/iceberg/blob/apache-iceberg-1.9.0/open-api/rest-catalog-open-api.yaml).
+
+!!! note
+
+    This implementation is designed against the IRC spec as of Iceberg release version 1.9.0.
+
+## Namespace Mapping
+
+An IRC server can be viewed as the root Lance namespace.
+The Iceberg multi-level namespaces map to the multi-level child namespaces.
+Whether the namespace is leveled and the number of levels depend on the specific IRC provider.
+
+## Table Definition
+
+A Lance table should appear as a table object in IRC in the shape of Iceberg [TableMetadata](https://github.com/apache/iceberg/blob/apache-iceberg-1.9.0/open-api/rest-catalog-open-api.yaml#L2482),
+with the following requirements:
+
+1. the [`location`](https://github.com/apache/iceberg/blob/apache-iceberg-1.9.0/open-api/rest-catalog-open-api.yaml#L2494) must point to the root location of the Lance table
+2. the [`properties`](https://github.com/apache/iceberg/blob/apache-iceberg-1.9.0/open-api/rest-catalog-open-api.yaml#L2499) must follow:
+    1. there is a key `table_type` set to `lance` (case insensitive)
```

Comment:
When a request comes in it seems we would need to know the type before we load the metadata?

Reply:
In my mind the IRC server should be able to manipulate what is returned as the response of LoadTable, for example if it's a Lance table, it can just generate a TableMetadata JSON payload without the need to read any actual Iceberg metadata JSON file, and provide this information in the properties section. Is that feasible?

Reply:
HMS with iceberg works like this. It look up the generic table object and then see what the `table_type` property is 

this also reminds me of Polaris generic table 
https://polaris.apache.org/in-dev/unreleased/generic-table/

Reply:
> it can just generate a TableMetadata JSON payload without the need to read any actual Iceberg metadata JSON file, and provide this information in the properties section. Is that feasible?

i think its feasible as long as the reader uses the `TableMetadata` and not re-read the `metadata.json` file (i think spark does this today).